### PR TITLE
Verify that all templates need to learn from the entire utterance.

### LINKED ISSUE #863: Integrated learner should only try to learn an utterance if it can fully understand the example
Per a discussion on July 1st with @charles-yang-upenn and @mitchmarcus we want to have our learner only try to learn new examples when it fully understands what is has been presented.

For example from a situation "Mom eats a cookie" we won't learn "eats cookie" or "mom eats" because that wouldn't be a complete match of the situation text. 

By limiting our learning to only these 'complete examples', which we believe kids do as well, we constrain our utterance space considerably and don't have to deal with the question of how to eventually filter out the "eats cookie" or "mom eats" examples which would always match the longer situation text of "mom eats a cookie".

### PR DESCRIPTION
**Pull request by LichMaster98**
Closes #863

### CODE REVIEW

File: adam/learner/attributes.py [hunk 1]
```diff
@@ -98,7 +99,25 @@ def _candidate_templates(
...
+            # We strictly ignore english determiners here for determining
+            # the length needed of our surface template
+            if covers_entire_utterance(
+                bound_surface_template,
+                language_concept_alignment,
+                token_sequence_count=len(
+                    [
+                        token
+                        for token in language_concept_alignment.language.as_token_sequence()
+                        if token not in ["a", "the"]
```
Comment (LichMaster98) [inline adam/learner/attributes.py:116]: @Reviewer - Any ideas why the attribute learner needs this specific condition to be able to learn color attributes while the relationship and verb learner can handle the determiners?

Comment (LichMaster98) [reply adam/learner/attributes.py:116]: This was found as a result of the attribute SurfaceTemplate not appropriately adding the SLOT variables as having determiners.

File: adam/learner/learner_utils.py [hunk 2]
```diff
@@ -191,3 +195,37 @@ def sort_by_num_nodes(g: DiGraph) -> int:
...
+    for element in bound_surface_template.surface_template.elements:
+        if isinstance(element, str):
+            num_covered_tokens += 1
+        else:
+            num_covered_tokens += len(
+                language_concept_alignment.node_to_language_span[
+                    bound_surface_template.slot_to_semantic_node[element]
+                ]
+            )
+    # We may need to
```
Comment (gabbard) [inline adam/learner/learner_utils.py:216]: I think this comment is missing something

File: adam/learner/learner_utils.py [hunk 3]
```diff
@@ -191,3 +195,37 @@ def sort_by_num_nodes(g: DiGraph) -> int:
...
+    sized_tokens = (
+        len(language_concept_alignment.language.as_token_sequence())
+        if not ignore_determiners
+        else len(
+            [
+                token
+                for token in language_concept_alignment.language.as_token_sequence()
+                if token not in ["a", "the"]
```
Comment (gabbard) [inline adam/learner/learner_utils.py:224]: Do we have a "constant" for the determiner set somewhere?  I think we also somewhere track English-specific places in the code (e.g. on an issue). If so, add this there.

Comment (LichMaster98) [reply adam/learner/learner_utils.py:224]: Added: https://github.com/isi-vista/adam/issues/498#issuecomment-654942842

**Pull request merged**